script.js

```diff
@@ -25,33 +25,37 @@ class AudioController {
         this.stopMusic();
         this.victorySound.play();
     }
-    gameover() {
+    gameOver() {
         this.stopMusic();
-        this.gameoverSound.play();
+        this.gameOverSound.play();
     }
 }
 
 class MixOrMatch {
     constructor(totalTime, cards) {
         this.cardsArray = cards;
+        this.totalTime = totalTime;
         this.timeRemaining = totalTime;
-        this.countdown = null;
-        this.cardClickCount = 0;
-        this.busy = false;
         this.timer = document.getElementById('time-remaining')
         this.ticker = document.getElementById('flips');
-        this.totalClicks = 0;
-        this.cardToCheck = null;
-        this.matchedCards = [];
         this.audioController = new AudioController();
     }
 
     startGame() {
-        this.audioController.startMusic();
-        this.shuffleCards(this.cardsArray);
+        this.totalClicks = 0;
+        this.timeRemaining = this.totalTime;
+        this.cardToCheck = null;
+        this.matchedCards = [];
+        this.busy = true;
+        setTimeout(() => {
+            this.audioController.startMusic();
+            this.shuffleCards(this.cardsArray);
+            this.countdown = this.startCountdown();
+            this.busy = false;
+        }, 500)

```

Comment:
I kept the cards flipped up after the game ends since it looks nicer to the user, so I had to add a slight delay before shuffling them as to not give away their location.